lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -2735,35 +2756,41 @@ LogicalResult impl::validateUnaryOpArguments(ValueRange operands,
 
 FIRRTLType AsSIntPrimOp::inferUnaryReturnType(FIRRTLType input,
                                               Optional<Location> loc) {
-  int32_t width = input.getBitWidthOrSentinel();
-  if (width == -2) {
-    if (loc)
-      mlir::emitError(*loc, "operand must be a scalar type");
-    return {};
+  auto base = input.dyn_cast<FIRRTLBaseType>();
+  if (base) {
+    int32_t width = base.getBitWidthOrSentinel();
+    if (width != -2)
+      return SIntType::get(input.getContext(), width);
   }
-  return SIntType::get(input.getContext(), width);
+  if (loc)
+    mlir::emitError(*loc, "operand must be a scalar type");
+  return {};
 }
 
 FIRRTLType AsUIntPrimOp::inferUnaryReturnType(FIRRTLType input,
                                               Optional<Location> loc) {
-  int32_t width = input.getBitWidthOrSentinel();
-  if (width == -2) {
-    if (loc)
-      mlir::emitError(*loc, "operand must be a scalar type");
-    return {};
+  auto base = input.dyn_cast<FIRRTLBaseType>();
```

Comment:
Do we need `dyn_cast` here ?

Reply:
Wasn't sure if failing with error was better or assert.  I'll move to `cast`, non-base types shouldn't be possible here.  Good call, thanks!

Reply:
Did this as well for the other width inference methods, d'oh.  Thanks for the suggestion!